### PR TITLE
Edit Site: Update template navigation to use new link control.

### PR DESCRIPTION
**Pull request by epiqueras**
## Description

#19141 was merged with a feature to navigate to the template that an internal link will use to render its front end, using the `URLPopover`.

The `URLPopover` has since been replaced with the `LinkControl`. This PR moves the slot/fill extension introduced by #19141 to the new `LinkControl` component.

## How has this been tested?

It was verified that the feature works as expected with the new `LinkControl`.

## Screenshots

image

## Checklist:

- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by shaunandrews** [comment]
Whats the use-case in which I'd want to edit a Template from within the context of the link editor?

**Comment by epiqueras** [comment]
That's not what this is doing.

https://github.com/WordPress/gutenberg/issues/19204#issuecomment-572046016

**Review by Addison-Stavlo — COMMENTED** [review]
This is cool!

So I can go to the Site Editor and edit a template-part.  From here I can add a link to the front-page template (link defined to `/`), and then navigate to that template in the site editor via that link.  Fantastic! 🎉 

What I am having issues with is creating a second template (call it 'second-page') in here and adding a link to that second template (link defined to lead to `/second-page` ?):
1.  It seems to go to the front-page template either way, 
2. and that second template does not seem to persist in the Site Editor's list of templates when I go back into the Site Editor later (although it did get added to the list of 'templates' in the 'appearance' section).  

Are these behaviors expected at this point and/or represent future work to be done in follow ups?

**Comment by epiqueras** [comment]
>It seems to go to the front-page template either way,

`second-page` is not in the template hierarchy. These links go to the template the linked URL would render in.

>and that second template does not seem to persist in the Site Editor's list of templates when I go back into the Site Editor later (although it did get added to the list of 'templates' in the 'appearance' section).

It needs to be published to be part of the hierarchy.

**Comment by Addison-Stavlo** [comment]
Thanks for the info!

> It needs to be published to be part of the hierarchy.

If I go to "Appearance" -> "Templates" (or similarly "Template Parts"), there are Templates / Template Parts that are listed as "Published" that do not appear in the selection dropdown in the Top-Left of the site editor.  Is there something else I need to do for them to be "published" where the Site Editor is concerned?

> These links go to the template the linked URL would render in.

I am struggling to find how to set up a second page to render another template in for the sake of testing this nifty template navigation even further.  Do you happen to have any other info on this?  I just got done going through the most recent gutenbrerg times demo with no luck, is there anything else that may highlight this?

**Comment by epiqueras** [comment]
>If I go to "Appearance" -> "Templates" (or similarly "Template Parts"), there are Templates / Template Parts that are listed as "Published" that do not appear in the selection dropdown in the Top-Left of the site editor. Is there something else I need to do for them to be "published" where the Site Editor is concerned?

They need to be part of the resolved hierarchy somehow.

>I am struggling to find how to set up a second page to render another template in for the sake of testing this nifty template navigation even further. Do you happen to have any other info on this? I just got done going through the most recent gutenbrerg times demo with no luck, is there anything else that may highlight this?

Make a `single` template and link to a single post.

**Comment by Addison-Stavlo** [comment]
> Make a `single` template 

So I made a `single.html` template in the `demo-block-templates` folder, and now 'single' shows up in the site editor's template list.

> and link to a single post.

added a link in the header to `/?p=1`.  The slotfill for 'Edit Page Template' does not appear.  However, on the front-end `/?p=1` does properly load using the `single.html` template.  🤔

**Comment by epiqueras** [comment]
There was a bug with URLs with query strings. I just fixed it, and it works well now.

Thanks for catching it!

File: packages/block-editor/src/components/url-popover/link-viewer.js [hunk 1]
```diff
@@ -7,12 +7,9 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { pencil } from '@wordpress/icons';
-import { createSlotFill, ExternalLink, Button } from '@wordpress/components';
+import { ExternalLink, Button } from '@wordpress/components';
 import { safeDecodeURI, filterURLForDisplay } from '@wordpress/url';
-import { useMemo } from '@wordpress/element';
-
-const { Slot, Fill } = createSlotFill( 'BlockEditorURLPopoverLinkViewer' );
```
Comment (Addison-Stavlo) [inline packages/block-editor/src/components/url-popover/link-viewer.js]: If the `URLPopover` has been replaced, why do we need to make changes to this file?

Could there be any cases where this is still used where removing this slot may cause issues?

**Comment by Addison-Stavlo** [comment]
This is working great!  

I am about ready to approve, but just have the question about about the `link-viewer` file above.

File: packages/block-editor/src/components/url-popover/link-viewer.js (hunk 1 above)
Comment (epiqueras) [reply packages/block-editor/src/components/url-popover/link-viewer.js]: This was only added for this, so I am reverting it here before anyone gets a chance to use it.

**Comment by epiqueras** [comment]
Answered :smile:

**Review by Addison-Stavlo — APPROVED** [review]
This works great and is a really awesome feature to have working in the Site Editor!

Thanks for all your effort 😄

**Pull request merged**